File: GameAPI/src/main/java/model/Track.java
package model;

import dev.morphia.annotations.*;
import org.bson.types.ObjectId;
import org.json.JSONObject;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.Getter;
import lombok.Setter;
import java.util.HashMap;

@Getter
@Setter
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity("tracks")

public class Track {
    @Id
    private String id;

    private String name;

    private String planetId;

    private String seed;

    private HashMap<String,Integer> times;

    // region Getters and Setters

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getPlanetId() {
        return planetId;
    }

    public void setPlanetId(String planetId) {
        this.planetId = planetId;
    }

    public String getSeed() {
        return seed;
    }

    public void setSeed(String seed) {
        this.seed = seed;
    }

    public HashMap<String, Integer> getTimes() {
        return times;
    }

    public void setTimes(HashMap<String, Integer> times) {
        this.times = times;
    }

    // endregion

    public JSONObject toJSON() {
        return new JSONObject(this);
    }

}

File: UniverseUpdater/source/model/track.py
##The track object/DTO
class Track:

    #
    # times: Dictionary of username and their track times in milliseconds
    def __init__(self, name, planetId, seed=None, times=None, data=None ):
        self._id = None
        self.name = name
        self.planetId =  planetId
        self.data = data
        self.seed = seed
        self.times = times if times is not None else {}


    def __str__(self):
        dataString = "None" if self.data is None else (str(len(self.data)) + " bytes")
        seedString = "None" if self.seed is None else str(self.seed)
        return f"Track( id={self._id}, name='{self.name}', planetId={self.planetId}, seed={seedString}, data={dataString} )"


    def setData(self,data):
        self.data = data


    def setId(self,id):
        self._id=id
